tabbycat/results/scoresheet.py

```diff
@@ -39,13 +31,34 @@ def identical(self, other):
         """Base implementation. Does nothing."""
         return True
 
+    def winners(self):
+        """Returns ['aff'] is the affirmative team won, and ['neg'] if the negative
+        team won. `self._get_winners()` must be implemented by subclasses."""
+        if not self.is_complete():
+            return set()
+        return self._get_winners()
+
+    # Default methods
+    # Winners may be declared by the form as validation; by default do nothing.
+
+    def set_declared_winners(self, winners):
+        pass
+
+    def add_declared_winner(self, winner):
+        pass
+
+    def get_declared_winners(self):
+        pass
+
```

Comment:
I actually think I'd prefer that these classes explicitly fail when a caller tries to set or get a declared winner when it's not meant to, rather than fail silently 🙂

This will mean that at the form level, forms will have to take care to distinguish between when a declared winner is and isn't required, and not attempt to declare one if it's not (i.e., if it's collecting it only for form validation). I think I'm more okay with this than I am with providing a method that depending on circumstance sometimes silently does nothing. But there might be something you've noticed that I haven't considered, so let me know if you've found otherwise 🙂